### ISSUE TITLE
cylc ping passphrase problem

### ISSUE THREAD

**Issue by matthewrmshin** [cylc/cylc#1699]
Consider 2 users on 2 suite hosts on a shared file system. If user A runs a suite on host H1 and user B runs a suite on host H2 on the same port number, we can get into trouble with `cylc ping`.

If user A logs onto H2 and runs `cylc ping`, the client will get a Pyro `ConnectionDeniedError` and user B will get a message like this in her suite log:

```
2015-12-14T14:00:41Z WARNING - [client-connect] DENIED user-a@h2:cylc-ping ab6e4dd3-cbbf-4e4e-9349-e18a2105ad8c
```

Digging a bit further, it looks like the client is failing because it is trying to connect with a passphrase, which will be wrong.

However, the logic in `cylc.network.suite_info` (`SuiteInfoServer.get`) suggests that any command with the string `ping` is supposed to be free info (is this a potential bug in the future?), so really we should not need a passphrase to do `ping suite`.

(On the other hand, `ping task` probably should require a higher access privilege?)

**Comment by hjoliver** [comment]
I'm not seeing this problem - tested running two different suites on the same port on two hosts that see the same filesystem, both suites get pinged fine from both hosts.

Or did you mean running the same suite, or two suites with the same name, on the two hosts?

**Comment by hjoliver** [comment]
We currently assume that if `$HOME/.cylc/ports/SUITE` exists on host H it implies that SUITE is  running on host H, and the port file contains the port number that it is listening on.

Therefore, on a shared filesystem where port files are visible on multiple hosts, it is up to the user to determine what host a suite is running on before attempting to connect to it, otherwise her cylc clients will try to connect to the right port on the wrong host, and - as in your problem report - there may be another suite running there which will naturally reject these connection attempts.

The reason for the seemingly odd connect-with-passphrase behaviour is that the client initially attempts to get the ping info with no passphrase and is rejected because the user and suite names are wrong.  This *could be* because it is a pre cylc-6.5 (authentication upgrade) suite daemon, so it attempts to connect with available passphrases, which the target suite naturally rejects.  This logic can be exposed with `cylc ping --debug SUITE`. 

So, in the current implementation this is arguably just user error - users on shared filesystems need to remember (or determine with `cylc scan`) where there suites are running.  That said, we can fix it, probably by recording the suite hostname in the port file as well as the port number.

In other words, this issue is really a duplicate of #1285.  It is certainly another symptom of the same problem, although the description there only considers the effect on an individual user.

**Comment by matthewrmshin** [comment]
@hjoliver Fair enough. Are you still working on #1285? Or should I have a go at it?

**Comment by hjoliver** [comment]
I'll have a look today at where I got to on this...

**Comment by hjoliver** [comment]
Closing as duplicate; further discussion on the original ticket #1285.
